packages/admin/dashboard/src/routes/orders/order-balance-settlement/components/order-balance-settlement-form/order-balance-settlement-form.tsx

```diff
@@ -0,0 +1,398 @@
+import { zodResolver } from "@hookform/resolvers/zod"
+import { AdminOrder, AdminPayment } from "@medusajs/types"
+import {
+  Button,
+  clx,
+  CurrencyInput,
+  Divider,
+  Input,
+  Label,
+  RadioGroup,
+  Select,
+  Textarea,
+  toast,
+} from "@medusajs/ui"
+import { useEffect, useMemo, useState } from "react"
+import { formatValue } from "react-currency-input-field"
+import { useForm } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+import * as zod from "zod"
+import { Form } from "../../../../../components/common/form"
+import { RouteDrawer, useRouteModal } from "../../../../../components/modals"
+import { KeyboundForm } from "../../../../../components/utilities/keybound-form"
+import {
+  useCreateOrderCreditLine,
+  useRefundPayment,
+} from "../../../../../hooks/api"
+import { currencies } from "../../../../../lib/data/currencies"
+import { formatCurrency } from "../../../../../lib/format-currency"
+import { getLocaleAmount } from "../../../../../lib/money-amount-helpers"
+import { getPaymentsFromOrder } from "../../../../../lib/orders"
+
+const OrderBalanceSettlementSchema = zod.object({
+  settlement_type: zod.enum(["credit_line", "refund"]),
+  refund: zod
+    .object({
+      amount: zod.string().or(zod.number()).optional(),
+      note: zod.string().optional(),
+    })
+    .optional(),
+  credit_line: zod
+    .object({
+      amount: zod.string().or(zod.number()).optional(),
+      reference: zod.string().optional(),
+      reference_id: zod.string().optional(),
+      note: zod.string().optional(),
+    })
+    .optional(),
+})
+
+export const OrderBalanceSettlementForm = ({
+  order,
+}: {
+  order: AdminOrder
+}) => {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+  const [activePayment, setActivePayment] = useState<AdminPayment | null>(null)
+  const payments = getPaymentsFromOrder(order)
+  const pendingDifference = order.summary.pending_difference * -1
+
+  const form = useForm<zod.infer<typeof OrderBalanceSettlementSchema>>({
+    defaultValues: {
+      settlement_type: "refund",
+      refund: {
+        amount: 0,
+      },
+      credit_line: {
+        amount: 0,
+      },
+    },
+    resolver: zodResolver(OrderBalanceSettlementSchema),
+  })
+
+  const { mutateAsync: createCreditLine, isPending: isCreditLinePending } =
+    useCreateOrderCreditLine(order.id)
+
+  const { mutateAsync: createRefund, isPending: isRefundPending } =
+    useRefundPayment(order.id, activePayment?.id!)
+
+  const settlementType = form.watch("settlement_type")
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    if (data.settlement_type === "credit_line") {
+      await createCreditLine(
+        {
+          amount: parseFloat(data.credit_line!.amount! as string) * -1,
```

Comment:
thought: I might have missed a critical detail in a previous PR–I would have expected credit lines to always have a positive value. Negative values would have a significantly different meaning.

For example:

1. Customer places order
```
Sale: $100 (debit)
Credit line: $0
Net total: $100
```
2. Customer requests return (and refund)
```
Sale: $100 (debit)
Credit line: $100
Net total: $0 (debit - credit)
```

Let me know if I am missing something? I haven't spotted this earlier, so it's likely that I am missing some context for why it's created as a negative value here.

Reply:
We add negative credit lines to refund to store credit. Otherwise we need a different way to indicate that a credit line "takes from the balance" or "adds to the balance".

When its a refund, we take from the balance through the negative credit line. When its a "payment", we add to the balance through a positive credit line. This is exactly how we do for payments/refunds as well. 

Here is an example of an order refunded partially through payment method and the other through store credit

<img width="1039" alt="Screenshot 2025-03-31 at 23 21 54" src="https://github.com/user-attachments/assets/dabee7a1-e4c9-4b02-97d2-4c837312a282" />

Reply:
> When its a "payment", we add to the balance through a positive credit line

Is it correctly understood that this used when paying with store credit? Or what does "payment" refer to here?

Reply:
Yes exactly! When we apply store credit to cart, we create a positive credit line.